src/item.cpp

```diff
@@ -1519,19 +1519,19 @@ double item::effective_dps( const Character &guy, monster &mon ) const
     // sum average damage past armor and return the number of moves required to achieve
     // that damage
     const auto calc_effective_damage = [ &, moves_per_attack]( const double num_strikes,
-    const bool crit, const Character & guy, monster & mon ) {
-        monster temp_mon = mon;
+    const bool crit, const Character & guy, Creature & mon ) {
+        Creature *temp_mon = &mon;
```

Comment:
Why did this become a pointer instead of a reference?  I can't see a reason, but I'm probably missing something.  Could you add a comment in the code explaining why?

Reply:
this was a change in my other PR and i missed the window to edit it before it got merged. this PR doesn't touch that file atm, but i can make sure to note down to go back and re-evaluate why i did that.